jrt/src/com/yahoo/jrt/MaybeTlsCryptoSocket.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.jrt;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+
+/**
+ * A crypto socket for the server side of a connection that
+ * auto-detects whether the connection is tls encrypted or unencrypted
+ * using clever heuristics. The assumption is that the client side
+ * will send at least one RPC request before expecting anything from
+ * the server. The first 9 bytes are inspected to see if they look
+ * like part of a tls handshake or not (RPC packet headers are 12
+ * bytes).
+ **/
+public class MaybeTlsCryptoSocket implements CryptoSocket {
+
+    private static final int snoop_size = 9;
```

Comment:
Consider following the convention of naming constants with upper-case.

Reply:
oops, c++ leak